third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -2840,6 +2879,35 @@ func cidrOrSizeDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	return strings.HasPrefix(new, "/") && strings.HasSuffix(old, new)
 }
 
+// We want to suppress diffs for the default auto provisioning defaults
+func containerClusterAutoProvisioningDefaultSuppress(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Instead of a DSF, can we make the field O+C? I think that's more consistent with other `scopes` fields. Users can choose to not manage scopes or authoritatively manage them.

Reply:
I don't think we can make this O+C because it needs to append those defaults onto whatever is passed in.  If I make it O+C it shows a diff that it's trying to take the mandatory ones off (maybe that's what I should call them rather than defaults), and make them just the new ones.
If that makes sense.

Reply:
I don't believe they're actually mandatory, unless I'm missing something. If `logging_service` and `monitoring_service` are both set to `NONE`, GKE won't need those scopes to function. See the notes at https://www.terraform.io/docs/providers/google/r/container_cluster.html#oauth_scopes

Reply:
It turns out that even with these values set to `none`, `monitoring.write` is still set. That's a frustrating behaviour by the API!